models/RestrictedAdding.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Models;
+
+use Models\Exceptions\RestrictionNotPassedException;
+
+/**
+ * Adding model which only adds values if all of them are present in a
+ * predefined list.
+ */
+final class RestrictedAdding implements Adding
+{
+
+    /**
+     * Ctor.
+     * 
+     * @param Adding   $origin  original adding model.
+     * @param string[] $demands array of required fields.
+     */
+    public function __construct(
+        /**
+         * Original adding model.
+         *
+         * @var Adding
+         */
+        private Adding $origin,
+
+        /**
+         * Array of required fields.
+         *
+         * @var string[]
+         */
+        private array $demands
+    ) {
+    }
+
+    /**
+     * Only adds values if they all of values in demands are present in a
+     * given array of data.
+     * {@inheritDoc}
+     * @throws RestrictionNotPassedException if some values are not present.
+     */
+    public function added(array $data): static
+    {
+        if (
+            count(
+                array_intersect(
+                    array_keys($data),
+                    $this->demands
+                )
+            ) !== count($this->demands)
+        ) {
```

Comment:
If array with data does not contain keys with all names from demands then their intersection will have less values that demands.